CMakeLists.txt

```diff
@@ -168,7 +170,18 @@ if(CMAKE_COMPILER_IS_GNU)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wshadow")
     endif()
     if (GCC_VERSION VERSION_GREATER 5.0)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
+        CHECK_C_COMPILER_FLAG("-Wformat-signedness" C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
+        if(C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
+        endif()
+    endif()
+    CHECK_C_COMPILER_FLAG("-Wno-signed-one-bit-field" C_COMPILER_SUPPORTS_WSIGNED_ONE_BIT_FIELD)
+    if(C_COMPILER_SUPPORTS_WSIGNED_ONE_BIT_FIELD)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-signed-one-bit-field")
+    endif()
+    CHECK_C_COMPILER_FLAG("-Wno-sign-compare" C_COMPILER_SUPPORTS_WSIGN_COMPARE)
+    if(C_COMPILER_SUPPORTS_WSIGN_COMPARE)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-sign-compare")
```

Comment:
No, we do want `-Wsign-compare` enabled. It's usually a true positive.

Reply:
Got it. Fixed in 0be6aa9